divi/src/masternode.cpp

```diff
@@ -697,7 +697,22 @@ void CMasternodeBroadcastFactory::createWithoutSignatures(
                                     ? createDelayedMasternodePing(mnbRet)
                                     : CMasternodePing(txin));
     mnbRet.lastPing = mnp;
-    mnbRet.sigTime = mnp.sigTime;
+
+    /* We have to ensure that the sigtime of the broadcast itself is later
+       than when the collateral gets its 15 confirmations (and it might not
+       even be confirmed once yet).  */
+    const auto* pindexConf = mnbRet.GetCollateralMinConfirmationBlock();
+    if (pindexConf == nullptr) {
+        /* The signature time is accepted on the network up to one hour
+           into the future.  That will also be more than enough to be
+           later than the confirmation block time assuming the collateral
+           is going to be mined "now".  */
+        mnbRet.sigTime = GetAdjustedTime() + 60 * 60;
+        LogPrint("masternode", "Using future sigtime for masternode broadcast: %d\n", mnbRet.sigTime);
+    } else {
+        mnbRet.sigTime = pindexConf->GetBlockTime();
+        LogPrint("masternode", "Using collateral confirmation time for broadcast: %d\n", mnbRet.sigTime);
+    }
```

Comment:
Reduce the 1h offset to 45 minutes to allow the broadcast to be used right away without potential p2p clock drift issues.

Reply:
Good point, I've updated the time now to 45 minutes.